app/views/users/shared/_links.html.erb

```diff
@@ -1,5 +1,7 @@
 <%- if controller_name != 'sessions' %>
-  <%= link_to t("devise.links.login"), new_session_path(resource_name) %><br />
+  <% login = t("devise.links.login") %>
+  <% login = t("devise.links.login_with_provider") if ['new_with_provider', 'create_with_provider'].include? action_name %>
```

Comment:
maybe is sexier to write `if action_name.in? %w(new_with_provider create_with_provider)` :smile: